tests/implement/tests/vector_view.rs

```diff
@@ -0,0 +1,141 @@
+use std::vec;
+use test_implement::*;
+use windows::*;
+use Windows::Foundation::Collections::*;
+
+#[implement(
+    Windows::Foundation::Collections::IVectorView<T>,
+)]
+struct TestView<T>(Vec<T>)
+where 
+    T: ::windows::RuntimeType + 'static;
+
+#[allow(non_snake_case)]
+impl<T: ::windows::RuntimeType + 'static> TestView<T> {
+    /* GetAt returns the value at the given index, and panic if given an invalid index */
+    fn GetAt(&self, index : u32) -> Result<T> {
+        let vec_size = self.0.len() as u32;
+        assert!(index < vec_size, "Given index ({}) was out of bounds for the IVectorView (length {})", index, vec_size);
+        self.0.get(index as usize).cloned().ok_or_else(|| panic!())
+    }
+   
+    /* Size returns the length of the underlying vector */
+    fn Size(&self) -> Result<u32> {
+        let size : u32 = self.0.len() as u32;
+        Ok(size)
```

Comment:
```suggestion
        Ok(self.0.len() as _)
```

Type inference is helpful here.